front/app/utils/dateUtils.ts

```diff
@@ -189,31 +159,26 @@ export function toFullMonth(date: string, resolution: IResolution) {
     .format(resolution === 'month' ? 'MMMM YYYY' : 'MMMM DD, YYYY');
 }
 
-// This function is used to display event start/end times with weekday names
-export const getEventDateWithWeekdays = (
-  event: IEventData,
-  formatMessage: (
-    messageDescriptor: MessageDescriptor,
-    values?:
-      | {
-          [key: string]: string | number | boolean | Date;
-        }
-      | undefined
-  ) => string
-) => {
-  const startAtMoment = moment(event.attributes.start_at);
-  const endAtMoment = moment(event.attributes.end_at);
+// Function used to determine whether a dot should be shown after the day in short date formats
+// Currently only used for German. Other locales can be added if needed.
+export function showDotAfterDay(locale: Locale) {
+  return locale === 'de-DE';
+}
+
+// Function used to get the event dates in a localized string format
+export function getEventDateString(event: IEventData) {
+  const startMoment = moment(event?.attributes.start_at);
+  const endMoment = moment(event?.attributes.end_at);
+
   const isEventMultipleDays =
-    startAtMoment.dayOfYear() !== endAtMoment.dayOfYear();
-  const startAtWeekday = getDayName(startAtMoment.weekday());
-  const endAtWeekday = getDayName(endAtMoment.weekday());
-  return isEventMultipleDays
-    ? `${
-        startAtWeekday && formatMessage(startAtWeekday)
-      }, ${startAtMoment.format('LLL')} - ${
-        endAtWeekday && formatMessage(endAtWeekday)
-      }, ${endAtMoment.format('LLL')}`
-    : `${startAtMoment.format('LL')} • ${startAtMoment.format(
-        'LT'
-      )} - ${endAtMoment.format('LT')}`;
-};
+    startMoment.dayOfYear() !== endMoment.dayOfYear() ||
+    startMoment.year() !== endMoment.year();
```

Comment:
`startMoment.year() !== endMoment.year()`

Took me a while to get why this was added. Could you add a comment?

So to see if my reasoning correct: in case you'd have a (peculiar) one-year-long event, this would say we don't have a multi-day event, while we have one. The year check makes sure the check still passes for that case.

Reply:
That's correct :) I'll add a comment to explain :+1: